src/lib/site.js

```diff
@@ -24,35 +39,248 @@ export async function getSiteMetadata() {
+
+      if (typeof value === 'undefined') return;
+
+      metadata.og[property] = value;
+    });
+  }
+
+  // Twitter Properties
+  // Loop through Twitter properties that rely on a non-object value
+
+  if (pageMetadata.twitter) {
+    const twitterProperties = ['cardType', 'description', 'imageUrl', 'title', 'username'];
+
+    twitterProperties.forEach((property) => {
+      const pageTwitter = pageMetadata.twitter?.[property];
+      const pageOg = metadata.og[property];
+      const value = pageTwitter || pageOg;
+
+      if (typeof value === 'undefined') return;
+
+      metadata.twitter[property] = value;
+    });
+  }
+
+  // Article Properties
+  // Loop through article properties that rely on a non-object value
+
+  if (metadata.og.type === 'article' && pageMetadata.article) {
+    metadata.article = {};
+
+    const articleProperties = ['author', 'modifiedTime', 'publishedTime', 'publisher'];
+
+    articleProperties.forEach((property) => {
+      const value = pageMetadata.article[property];
+
+      if (typeof value === 'undefined') return;
+
+      metadata.article[property] = value;
+    });
+  }
+
+  return metadata;
 }
 
-export function removeLastTrailingSlash(url) {
-  if (typeof url !== 'string') return url;
-  return url.replace(/\/$/, '');
+/**
+ * helmetSettingsFromMetadata
+ */
+
+export function helmetSettingsFromMetadata(metadata = {}, options = {}) {
+  const { link = [], meta = [], setTitle = true } = options;
+
+  const settings = {
+    htmlAttributes: {
+      lang: metadata.language,
+    },
+  };
+
+  if (setTitle) {
+    settings.title = metadata.title;
+  }
+
+  settings.link = [
+    ...link,
+    {
+      rel: 'canonical',
+      href: metadata.canonical,
+    },
+  ].filter(({ href } = {}) => !!href);
+
+  settings.meta = [
+    ...meta,
+    {
+      name: 'description',
+      content: metadata.description,
+    },
+    {
+      property: 'og:title',
+      content: metadata.og?.title || metadata.title,
+    },
+    {
+      property: 'og:description',
+      content: metadata.og?.description || metadata.description,
+    },
+    {
+      property: 'og:url',
+      content: metadata.og?.url,
+    },
+    {
+      property: 'og:image',
+      content: metadata.og?.imageUrl,
+    },
+    {
+      property: 'og:image:secure_url',
+      content: metadata.og?.imageSecureUrl,
+    },
+    {
+      property: 'og:image:width',
+      content: metadata.og?.imageWidth,
+    },
+    {
+      property: 'og:image:height',
+      content: metadata.og?.imageHeight,
+    },
+    {
+      property: 'og:type',
+      content: metadata.og?.type || 'website',
+    },
+    {
+      property: 'og:site_name',
+      content: metadata.siteTitle,
+    },
+    {
+      property: 'twitter:title',
+      content: metadata.twitter?.title || metadata.og?.title || metadata.title,
+    },
+    {
+      property: 'twitter:description',
+      content: metadata.twitter?.description || metadata.og?.description || metadata.description,
```

Comment:
im going to leave this one off, feel free to make a PR for it you can get credit for it in the commit log :)

only thinking is i wonder if it should be 2 functions or do you think sanitize makes sense? as opposed to something like `sanitize` in addition to `trim`, i dont know 🤷 could probably put it/them in `lib/util` though to pull it in. it could probably be useful for other things in addition to helmet data too if you want to make it more generic, just a thought

Reply:
yeah good idea! Makes much more sense that way. I will prepare a PR for putting them at lib/util

Reply:
one thing i just noticed this morning is Yoast automatically trims the meta titles and it made me realize i wonder if we should leave the titles alone but we coudl trim the descriptions, but maybe only if they're not managed in Yoast?

For search engines the trimming makes sense, but when i sent the link on discord for instance, it would be nice to have the full thing

![image](https://user-images.githubusercontent.com/1045274/116699502-a5178200-a993-11eb-97ea-8535735aab7a.png)

additionally i think google will automatically just only use the first part right?

Reply:
So I think would be nice to have both functionalities:

1. **Remove extra spaces**: This function does not resolve anything critical because as far as I have tested both twitter cards and OG make this for you. I have tested some descriptions with break lines and extra spaces and they both (og and twitter) remove that, showing a "clean" text. The only reason to have this would be to have "cleaner" `head tags` at `html` (avoiding what is shown at my image above). The use case for this is limited to users **not** using Yoast: for example an author writing a bio using paragraphs (as it is allowed). I think can be a nice handle as it does not modify original text aside from removing more than two spaces together, and this way we assure that the resultant tags are being created sanitized. 

2. **Max length**: You are right, I realized all services already slice your text to match their specifications (also by adding the final '...', so the function above would result on 6 dots 😆 ). My original idea was to only use this one on each specific case, for example: 

```
    {
      property: 'twitter:title',
      content: metadata.title.slice(70)
    },
    {
      property: 'twitter:description',
      content: metadata.description.slice(200)
   },
   {
      property: 'og:description',
      content: metadata.description.slice(300)
   },
```
Again this does not resolve any problem, but this way the html head tags are controlled and contain exactly the info is required for each case instead of for example sending a 400 length description as `twitter:description` when in reality only half will be used.

Maybe is too specific ... what do you think @colbyfayock ?

Reply:
im thinking maybe we implement the first one and leave the 2nd one as is, mostly only because at least the platforms that can use the longer length can include them 🤷‍♂️